client/mm/mm.go

```diff
@@ -550,6 +633,56 @@ func (m *MarketMaker) botBalance(botID string, assetID uint32) uint64 {
 	return 0
 }
 
+func (m *MarketMaker) modifyBotCEXBalance(botID string, assetID uint32, amount uint64, increase bool) {
```

Comment:
Delta-based value tracking tends to be fragile. Much better to find an idempotent pattern, such that we track the server's reported balance and pending deposits/withdraws to calculate balance.

Reply:
The server's balance doesn't really tell us much because the balance of the CEX will be split between multiple bots. Also, not just deposits/withdrawals change the balance, but also trades. Maybe a better pattern would be to keep track of each of the actions which changed the balance and how much it changed it by. Then we could definitely avoid any double counting. Is this kind of what you're suggesting?

Reply:
I'll leave it up to you. Just wanted to point out the scary pattern.

Reply:
You're probably right, I'll think of something. The balance management for the DEX should be updated then too.